### DISCUSSION TITLE
how can I use your rewrite of Go WebRTC in chrome/firefox browsers?

### DISCUSSION THREAD

**Discussion by xujingzhou** [pion/webrtc#1753] [Q&A] [answered]
Did you rewrite Google's WebRTC entirely in Go? My question is: Chrome and Firefox already have native WebRTC (Google) built in, how can I use your rewrite of Go WebRTC in these browsers?

**Comment by Sean-Der** [comment, marked as answer]
Yes Pion WebRTC is a full implementation of WebRTC. We implemented the entire stack. We have pure Go packages for ICE, DTLS, SCTP, RTP, RTCP and the PeerConnection API. We didn't re-implement Google's implementation, Pion is a clean room implementation so you will not see any patterns/sections that look similar. We tried to make everything idiomatic Go and learned a lot along the way!

Using Pion in FireFox or Chrome is technically possible, but would be very hard. There is no standardized API for anything but Javascript, so will not be an easy switch. If you do start this project I would love to help! I think the project with the most likelihood of success would be putting [webrtc-rs](https://github.com/webrtc-rs/) in FireFox. There is already precedent for using Rust in FireFox, no one uses Go in the browser today.

The value of Pion today is mostly in building other projects, you can see them in [awesome-pion](https://github.com/pion/awesome-pion)
